src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -1256,6 +1256,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="CrossPlatformExecutable" />
   </ItemGroup>
 
+  <!-- Native AOT compilation -->
+  <ItemGroup>
+    <ProjectCapability Include="NativeAOTProperties" Condition="'$(ShowNativeAOTProperties)' == 'true'"/>
+  </ItemGroup>
+
+  <!-- Native AOT properties should be shown by default for projects targeting .NET 8 or higher, except for class libraries, WPF, WinForms projects -->
+  <PropertyGroup>
+    <ShowNativeAOTProperties Condition="'$(ShowNativeAOTProperties)' == ''
+                           and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                           and '$(_TargetFrameworkVersionWithoutV)' >= '8.0'">true</ShowNativeAOTProperties>
+    <ShowNativeAOTProperties Condition="'$(UseWPF)' == 'true'
```

Comment:
Do we want to check `'$(ShowNativeAOTProperties)' == ''`?

Reply:
I'd say yes, we want to check if the property has not been set; so it defaults to true.

Reply:
I think the logic here is that the second `ShowNativeAOTProperties` needs to be able to override the first, so checking `'$(ShowNativeAOTProperties)' == ''` on the second instance would break that.

Reply:
> I think the logic here is that the second `ShowNativeAOTProperties` needs to be able to override the first, so checking `'$(ShowNativeAOTProperties)' == ''` on the second instance would break that.

Then flip the logic around so the `UseWPF` and `UseWindowsForms` checks happen first.

Reply:
Got it, thanks for the clarification.